src/2.1.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright 2016 Andrew Vaziri
+// CtCI Question 2.1
+// |||Prompt|||
+// Write code to remove duplicates from an unsorted linked list.
+// |||Questions I might ask|||
+// --Do I have to keep the elements that are not removed in the same order?
+// --Can I say that the list holds integers whiteboard simplicity sake?
+// --Am I given the list by reference or by value? Do I return a list or a
+// pointer to a list?
```

Comment:
Another possibility: Return (and take?) an iterator, generating the output lazily.